### PR TITLE
[V3 Economy] Expand payday output

### PR DESCRIPTION
**Pull request by palmtree5**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
See #1371 for the original suggestion. This implements display of the user's current balance and (if applicable) leaderboard position when getting a payday.

### CODE REVIEW

File: redbot/cogs/economy/economy.py [hunk 1]
```diff
@@ -252,6 +252,27 @@ def __init__(self, bot: Red):
                         credits_name
                     )
                 )
+                await ctx.send(
+                    _("You currently have {} {}.".format(
```
Comment (tekulvw) [inline redbot/cogs/economy/economy.py]: The i18n object requires  `_("...").format()`

**Review by tekulvw — DISMISSED** [review]

File: redbot/cogs/economy/economy.py (hunk 1 above)
Comment (palmtree5) [reply redbot/cogs/economy/economy.py]: whoops, I'll fix that now

**Comment by Stradinius** [comment]
Thanks for committing this, palmtree!

File: redbot/cogs/economy/economy.py [hunk 2]
```diff
@@ -252,6 +252,27 @@ def __init__(self, bot: Red):
                         credits_name
                     )
                 )
+                await ctx.send(
+                    _("You currently have {} {}.").format(
+                        str((await bank.get_balance(author))), credits_name
+                    )
+                )
+                if await bank.is_global():
```
Comment (tekulvw) [inline redbot/cogs/economy/economy.py]: You're doing a `bank.is_global()` check inside of a `bank.is_global()` check here.

File: redbot/cogs/economy/economy.py [hunk 3]
```diff
@@ -252,6 +252,27 @@ def __init__(self, bot: Red):
...
+                    bank_sorted = sorted(await bank.get_global_accounts(),
+                                         key=lambda x: x.balance, reverse=True)
+                else:
+                    bank_sorted = sorted(await bank.get_guild_accounts(guild),
+                                         key=lambda x: x.balance, reverse=True)
+                topten = bank_sorted[:10]
+                pos = 1
+                for acc in topten:
```
Comment (tekulvw) [inline redbot/cogs/economy/economy.py]: Might be easier to do something like:

```py
names = [acc.name for acc in topten]
try:
    pos = names.index(author.display_name)
except ValueError:
    pass
else:
    await ctx.send(...)
```

**Review by tekulvw — DISMISSED** [review]

File: redbot/cogs/economy/economy.py (hunk 2 above)
Comment (palmtree5) [reply redbot/cogs/economy/economy.py]: good catch lol

**Comment by palmtree5** [comment]
So the leaderboard changes from #1435 have been implemented, but with some stuff written for this PR.

We now have two functions relating to a bank leaderboard

`get_leaderboard` which gets the leaderboard
and
`get_leaderboard_position` which gets the user's position on the leaderboard

**Comment by palmtree5** [comment]
This is tested and working fine as well

**Comment by Stradinius** [comment]
Not sure if this is already a topic covered in V3 or not, but when a user bets with [p] slot, and wins, it does not take the initial bet before paying out.  So a 4* win actually becomes a 5* win.  Since we're already covering the economy cog here, I figured this would be the best place to note that.

**Comment by tekulvw** [comment]
We've seen that discussion in #876 and decided to keep the existing functionality.

**Review by tekulvw — APPROVED** [review]

**Pull request merged**